### PR TITLE
Added supersonic Euler/NS test cases.

### PR DESCRIPTION
**Pull request by tarikdzanic**

### CODE REVIEW

**Review by WillTrojak — COMMENTED** [review]
Seems fine to me. Although for the phi function you might like to consider the min and max functions, as they are a bit cleaner and I imagine the shock capturing can handle the sharper jump.

**Comment by tarikdzanic** [comment]
I'm using Lobatto points so I'm a little worried about roundoff error causing some issues at the coincident points at the center

File: 2d-double-mach-reflection/double-mach-reflection.ini [hunk 1]
```diff
@@ -0,0 +1,87 @@
+[backend]
+precision = single
```
Comment (FreddieWitherden) [inline 2d-double-mach-reflection/double-mach-reflection.ini]: Make this double for consistency with other cases?

Comment (FreddieWitherden) [inline 2d-double-mach-reflection/double-mach-reflection.ini]: Okay to make this double?

Comment (tarikdzanic) [reply 2d-double-mach-reflection/double-mach-reflection.ini]: Fixed.

**Comment by tarikdzanic** [comment]
I will run the tests again today to make sure everything runs as expected and then I think it's good to merge

**Comment by FreddieWitherden** [comment]
Are there any pending action items here before we can merge?

**Comment by tarikdzanic** [comment]
Just reran the tests, everything looks good

**Comment by FreddieWitherden** [comment]
@pv101 Do these cases look okay to you?

**Comment by pv101** [comment]
I've just launched them on my Macbook Pro (2021 M1 Max). Partitioning into 8 and with 'OMP_NUM_THREADS=1 mpirun -np 8' the Euler case says it will take 1h20m and the NS case 6h50m. I think the idea is to have cases that can run in close to 1h on a Macbook Pro. So the Euler case is fine. Is there any scope for reducing the cost of the NS case?

**Comment by tarikdzanic** [comment]
I can reduce the resolution and increase the time step a bit, that should lower it down to around an hour or so.

**Comment by pv101** [comment]
Sounds good. Once pushed I'll give it a test.

**Comment by tarikdzanic** [comment]
Can you try these new settings and let me know the runtime?

**Comment by pv101** [comment]
1h 44m - I'm OK with that. @FreddieWitherden ?

File: 2d-double-mach-reflection/double-mach-reflection.ini [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+[soln-bcs-wall]
+type = slp-adia-wall
+
+[soln-plugin-nancheck]
+nsteps = 50
+
+[soln-plugin-writer]
+dt-out = 0.01
+basedir = .
+basename = doublemach-{t:.2f}
```
Comment (pv101) [inline 2d-double-mach-reflection/double-mach-reflection.ini]: Not consistent with what's in the docs for this case in the main repo?

File: 2d-viscous-shock-tube/viscous-shock-tube.ini [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+[soln-bcs-sym]
+type = slp-adia-wall
+
+[soln-plugin-nancheck]
+nsteps = 50
+
+[soln-plugin-writer]
+dt-out = 0.01
+basedir = .
+basename = viscous_shock_tube_2d-{t:.2f}
```
Comment (pv101) [inline 2d-viscous-shock-tube/viscous-shock-tube.ini]: Not consistent with what's in the docs for this case in the main repo?

**Comment by FreddieWitherden** [comment]
Do the cases also work in single precision?

**Comment by FreddieWitherden** [comment]
Also, do we need to compress the meshes?  Just makes them a bit more of a pain to import.

**Comment by tarikdzanic** [comment]
Cases work in both single and double precision. I can push the raw meshes instead of the compressed ones, let me know what you prefer.

**Comment by FreddieWitherden** [comment]
> Cases work in both single and double precision. I can push the raw meshes instead of the compressed ones, let me know what you prefer.

How big are the uncompressed meshes?

**Comment by tarikdzanic** [comment]
Double mach goes from 4 MB to 20 MB, viscous shock tube goes from 1 MB to 5 MB.

**Comment by FreddieWitherden** [comment]
> Double mach goes from 4 MB to 20 MB, viscous shock tube goes from 1 MB to 5 MB.

Thanks.  What does `xz -9` give?

**Comment by tarikdzanic** [comment]
About 3 times smaller

**Comment by FreddieWitherden** [comment]
We'll go with that then! Best way to uncompress is `xzcat`.

Are any parameter changes needed for single precision?  Just want to make it as fool-proof as possible for users.

**Comment by tarikdzanic** [comment]
The default parameters seem to be okay for both

**Comment by FreddieWitherden** [comment]
@pv101 Are we okay to merge here?

**Comment by pv101** [comment]
Yes.

**Pull request merged**